src/@types/Icon.ts

```diff
@@ -1,13 +1,28 @@
-import React from 'react'
+import React, { ReactNode } from 'react'
 import { SystemStyleObject } from '../../styled-system/types'
 
-interface Icon {
+interface IconType {
   cssRaw?: SystemStyleObject
   fill?: string
   height?: number
   size?: number
   style?: React.CSSProperties
   width?: number
+  animated?: boolean
+  animationComplete?: () => void
 }
 
-export default Icon
+export interface AnimatedIconType {
```

Comment:
I think you can use `interface AnimatedIconType extends IconType` here for inheritance.

By convention on this project we generally export a single, default export per file, with the filename matching the type name. So `IconType` would go in `IconType.ts` and `AnimatedIconType` would go in `AnimatedIconType.ts`.

Reply:
Good idea to use interface AnimatedIconType extending IconType. 
Also, I will separate the types into their own files following the project convention.